app/services/attribute_asserter.rb

```diff
@@ -15,10 +13,10 @@ class AttributeAsserter
     :phone
   ].freeze
 
-  def initialize(user, service_provider, authn_request)
-    self.user = user
-    self.service_provider = service_provider
-    self.authn_request = authn_request
+  URI_PATTERN = Saml::Idp::Constants::REQUESTED_ATTRIBUTES_CLASSREF
+
+  def initialize(options)
```

Comment:
ah so you'd rather use hash than ruby named args? I kind of prefer named args but not a big deal: https://robots.thoughtbot.com/ruby-2-keyword-arguments#keyword-arguments-vs-options-hash